csr_check_test.go

```diff
@@ -1992,6 +2355,177 @@ func TestAuthorizeServingRenewal(t *testing.T) {
 	}
 }
 
+func TestGetServingCert(t *testing.T) {
+	defaultPort := int32(25535)
+	defaultAddr := "127.0.0.1"
+	defaultNode := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Status: corev1.NodeStatus{
+			Addresses: []corev1.NodeAddress{
+				{Type: corev1.NodeInternalIP, Address: defaultAddr},
+			},
+			DaemonEndpoints: corev1.NodeDaemonEndpoints{
+				KubeletEndpoint: corev1.DaemonEndpoint{
+					Port: defaultPort,
+				},
+			},
+		},
+	}
+
+	wrongAddr := defaultNode.DeepCopy()
+	wrongAddr.Status.DaemonEndpoints.KubeletEndpoint.Port = int32(25544)
+
+	uninitialized := defaultNode.DeepCopy()
+	uninitialized.Status = corev1.NodeStatus{}
+
+	tests := []struct {
+		name      string
+		nodeName  string
+		node      *corev1.Node
+		rootCerts []*x509.Certificate
+		wantErr   string
+		serverErr error
+	}{
+		{
+			name:      "all good",
+			nodeName:  "test",
+			node:      defaultNode,
+			rootCerts: []*x509.Certificate{parseCert(t, rootCertGood)},
+		},
+		{
+			name:      "unknown certificate",
+			nodeName:  "test",
+			node:      defaultNode,
+			rootCerts: []*x509.Certificate{parseCert(t, differentCert)},
+			wantErr:   "x509: certificate signed by unknown authority",
+		},
+		{
+			name:      "node not found",
+			nodeName:  "test",
+			rootCerts: []*x509.Certificate{parseCert(t, rootCertGood)},
+			wantErr:   "nodes \"test\" not found",
+			serverErr: fmt.Errorf("nodes \"test\" not found"),
+		},
+		{
+			name:      "wrong address",
+			nodeName:  "test",
+			node:      wrongAddr,
+			rootCerts: []*x509.Certificate{parseCert(t, rootCertGood)},
+			wantErr:   "dial tcp 127.0.0.1:25544: connect: connection refused",
+		},
+		{
+			name:     "no pool provided",
+			nodeName: "test",
+			node:     wrongAddr,
```

Comment:
should this be `wrongAddr`? 

i might be slightly confused byt the test name/logic